.github/workflows/_static_analysis.yaml

```diff
@@ -0,0 +1,19 @@
+name: Static Analysis
+
+on:
+  workflow_call:
+
+jobs:
+  static-analysis:
+    name: Run all static analysis checks
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: newrelic/newrelic-infra-checkers@v1
+      - name: golangci-lint
+        uses: golangci/golangci-lint-action@v3
+        continue-on-error: ${{ github.event_name != 'pull_request' }}
```

Comment:
why do we have this? Does it always fail if it is not a PR?

Reply:
To be honest, I don't know.  Maybe for some historical reason I'm not aware of... I can remove it and add it back in case we face any issues.

Reply:
it makes sense!